framework/tests/value/Switch.java

```diff
@@ -138,4 +151,25 @@ void testStrings(String s) {
         }
         @StringVal({"Good", "Bye", "Hello"}) String q = s;
     }
+
+    void testStrings2(String s) {
+        String a;
+        switch (a = s) {
+            case "Good":
+                @StringVal("Good") String x1 = a;
+                @StringVal("Good") String x2 = s;
+            case "Bye":
+                @StringVal({"Good", "Bye"}) String y1 = a;
+                @StringVal({"Good", "Bye"}) String y2 = s;
+                break;
+            case "Hello":
+                @StringVal("Hello") String z1 = a;
+                @StringVal("Hello") String z2 = s;
+                break;
```

Comment:
Can you add a test, for ints, Strings or something else, where a case falls through into the default case?